minecraft/forge/src/main/java/dev/kosmx/playerAnim/forge/ForgeClientEvent.java

```diff
@@ -5,31 +5,31 @@
 import dev.kosmx.playerAnim.minecraftApi.PlayerAnimationRegistry;
 import net.minecraft.server.packs.resources.ResourceManagerReloadListener;
 import net.neoforged.api.distmarker.Dist;
-import net.neoforged.fml.common.EventBusSubscriber;
+import net.neoforged.bus.api.IEventBus;
 import net.neoforged.neoforge.client.event.RegisterClientReloadListenersEvent;
-import net.neoforged.bus.api.SubscribeEvent;
 import net.neoforged.fml.ModList;
 import net.neoforged.fml.common.Mod;
-import net.neoforged.fml.event.lifecycle.FMLClientSetupEvent;
 import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
-@EventBusSubscriber(modid = "playeranimator", bus = EventBusSubscriber.Bus.MOD, value = Dist.CLIENT)
+@Mod(value = "playeranimator", dist = Dist.CLIENT)
 public class ForgeClientEvent {
+    public static final Logger LOGGER = LoggerFactory.getLogger("player-animator");
+
+    public ForgeClientEvent(IEventBus bus) {
+        bus.addListener(this::resourceLoadingListener);
 
-    @SubscribeEvent
-    public static void clientSetup(FMLClientSetupEvent event) {
         if (Helper.isBendEnabled() && ModList.get().isLoaded("skinlayers3d")) {
             try {
-                SkinLayersTransformer.init(ForgeModInterface.LOGGER);
+                SkinLayersTransformer.init(LOGGER);
```

Comment:
This is intentionally setup event and not class ctor

Reply:
why

Reply:
mod class ctor will be invoked way earlier than seteup event.
doing some event subscribe from ctor is okay, but it is adviced, not to do any real initialization/etc from the ctor. The mod setup event is for that

Reply:
The other reason why you don't delete the ForgeModInterface.java:
This library contains platform independent code, modders must be able to load this on server-side.
If the ForgeModInterface is client-only, Forge will simply fail to load it on server.

And the setup event must not run on server